### PR TITLE
chore: rework API directory to use react-query

### PR DESCRIPTION
**Pull request by raastinn**
## Compatiblity

Any changes that require a newer version of the API are considered breaking
changes.

Common breaking changes:
 - Making a request to a new API endpoint that will return `403` or `404` in older versions of the API.
 - No longer sending request fields required by previous versions of the API 
 - Requiring new fields from an API endpoint that will not be present in
   responses from older versions of the API
 - Breaking backwards compatibility with old response data shapes

Choose one:
 - [x] The changes do not break compatibility
 - [ ] The changes potentially break compatibility

## Pre-Review Checklist
 - [x] All changes are tested
 - [x] All touched code documentation is updated
 - [x] All tests pass in your local environment
 - [x] Deepsource issues have been reviewed and addressed
 - [x] Debugging logging and commented out code has been removed

 
## Screenshots:
_Only required for visual changes_.
Screenshot from 2024-05-14 14-40-09
Screenshot from 2024-05-14 14-40-23

Also visually updated the API keys list to match the current standards as well

### CODE REVIEW

**Comment by ReeceHoffmann** [comment]
Just checked out a copy to see how it works. I would suggest switching the sections over to regular `BoxGroupSections`. The all blue highlighting doesn't make much sense now imo given that only the checkbox is interactable.

File: src/js/base/BoxGroupSectionSelect.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-type SelectBoxGroupSectionProps = {
+type SelectBoxGroupSectionProps = BoxGroupSectionProps & {
     active?: boolean;
     theme: DefaultTheme;
 };
 
 export const SelectBoxGroupSection = styled(BoxGroupSection)<SelectBoxGroupSectionProps>`
     background-color: ${props => (props.active ? props.theme.color.blue : "transparent")};
     color: ${props => (props.active ? props.theme.color.white : "inherit")};
-    cursor: pointer;
```
Comment (ReeceHoffmann) [inline src/js/base/BoxGroupSectionSelect.ts]: Why this change?

File: src/js/base/BoxGroupSectionSelect.ts [hunk 2]
```diff
@@ -1,17 +1,16 @@
 import styled, { DefaultTheme } from "styled-components";
-import { BoxGroupSection } from "./BoxGroupSection";
+import { BoxGroupSection, BoxGroupSectionProps } from "./BoxGroupSection";
 import { CheckboxLabel } from "./CheckboxLabel";
 import { StyledCheckbox } from "./styled/StyledCheckbox";
 
-type SelectBoxGroupSectionProps = {
+type SelectBoxGroupSectionProps = BoxGroupSectionProps & {
```
Comment (ReeceHoffmann) [inline src/js/base/BoxGroupSectionSelect.ts]: Hmm feel like it makes more sense to either keep them separate or fuse them fully. Everything in `SelectBoxGroupsSecctionProps` is already in `BoxGroupSectionProps`

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]

File: src/js/base/BoxGroupSectionSelect.ts (hunk 1 above)
Comment (raastinn) [reply src/js/base/BoxGroupSectionSelect.ts]: Initially made this change because `BoxGroupSection` already makes the cursor a pointer when there is an `onClick` and all of the `SelectBoxGroupSection` usages had an `onClick` I believe, so I found it redundant and since I changed the `onClick` to `Checkbox` I thought t shouldn't show a pointer for the whole item just the clickable area. I'll revert the changes made to this component since we don't use it in this PR anymore

**Review by ReeceHoffmann — APPROVED** [review]
